data/json/mapgen/abandoned_barn.json

```diff
@@ -89,7 +89,7 @@
         "|": "t_wall_wood"
       }
     },
-    "om_terrain": "desolatebarn",
```

Comment:
Changing id of overmap terrains will cause issues when loading saved game until you provide a migration code (see https://github.com/CleverRaven/Cataclysm-DDA/blob/b8ef478f8998929fc3ab53c9a24ccbbece595cb3/src/savegame.cpp#L317-L711).